### PR TITLE
SLVS-1271 Add AnalysisListener.RaiseHotspots implementation

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Fixes #

### CODE REVIEW

File: src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        raiseFindingParamsToAnalysisIssueConverter.Received().GetAnalysisIssues(fileUri, Arg.Is<IEnumerable<RaisedFindingDto>>(x => !x.Any()));
+        analysisService.Received().PublishIssues(fileUri.LocalPath, analysisId, Arg.Is<IEnumerable<IAnalysisIssue>>(x => !x.Any()));
+    }
+
+    [TestMethod]
+    public void RaiseFindings_NoSupportedLanguages_PublishesEmpty()
+    {
+        var analysisIssue1 = CreateAnalysisIssue("csharpsquid:S100");
+        var analysisIssue2 = CreateAnalysisIssue("csharpsquid:S101");
+        var findings = new[] { analysisIssue1, analysisIssue2 };
```
Comment (gabriela-trutan-sonarsource) [inline src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs:98]: Please remove this unused variable

**Review by gabriela-trutan-sonarsource — APPROVED** [review]
I have some minor feedback

File: src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        var findingsByFileUri = new Dictionary<FileUri, List<TestFinding>> { { new FileUri("file://C:/somefile"), raisedFindings } };
+        var raiseFindingParams = new RaiseFindingParams<TestFinding>("CONFIGURATION_ID", findingsByFileUri, true, Guid.NewGuid());
+
+        var analysisService = Substitute.For<IAnalysisService>();
+
+        var testSubject = CreateTestSubject(analysisService: analysisService);
+
+        testSubject.RaiseFinding(raiseFindingParams);
+
+        analysisService.DidNotReceiveWithAnyArgs().PublishIssues(default, default, default);
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs:267]: False positive. https://nsubstitute.github.io/help/received-calls/#ignoring-arguments

File: src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    {
+        var analysisStatusNotifierFactory = Substitute.For<IAnalysisStatusNotifierFactory>();
+        analysisStatusNotifier = Substitute.For<IAnalysisStatusNotifier>();
+        analysisStatusNotifierFactory.Create(nameof(SLCoreAnalyzer), filePath, analysisId).Returns(analysisStatusNotifier);
+        return analysisStatusNotifierFactory;
+    }
+
+    private TestFinding CreateTestFinding(string ruleKey)
+    {
+        return new TestFinding(default, default, ruleKey, default, default, default, default, default, default, default, default, default, default, default, default);
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs:310]: False positive. https://nsubstitute.github.io/help/received-calls/#ignoring-arguments

File: src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+        analysisService.Received().PublishIssues(fileUri.LocalPath, analysisId, Arg.Is<IEnumerable<IAnalysisIssue>>(x => !x.Any()));
+        analysisStatusNotifier.AnalysisFinished(0, TimeSpan.Zero);
+    }
+
+    [TestMethod]
+    public void RaiseFindings_NoKnownLanguages_PublishesEmpty()
+    {
+        var analysisIssue1 = CreateAnalysisIssue("csharpsquid:S100");
+        var analysisIssue2 = CreateAnalysisIssue("csharpsquid:S101");
+        var findings = new[] { analysisIssue1, analysisIssue2 };
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs]: Done

File: src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.Listeners.UnitTests/Implementation/Analysis/RaisedFindingProcessorTests.cs]: Done

**Pull request merged**